### ISSUE TITLE
filterDefaultValues are shared between <List />

### ISSUE THREAD

**Issue by AkselsLedins** [marmelab/react-admin#1983]
**What you were expecting:**

When switching between resources in the menu, the `filterDefaultValues` prop should not be merged.

**What happened instead:**
The filter prop is shared and thus during navigation they are stacking between the different resources.

**Steps to reproduce:**

1) Create a `<List>` of  `A`
![selection_031](https://user-images.githubusercontent.com/18367406/42156939-ad05521a-7dec-11e8-87d1-923513431c1d.png)

2) Create a `<List>` of  `B`
![image](https://user-images.githubusercontent.com/18367406/42156914-98ec37a8-7dec-11e8-9608-f9cd39d884a0.png)

3) Go to `/A`
This will trigger this query:
![selection_032](https://user-images.githubusercontent.com/18367406/42157003-dc99f788-7dec-11e8-945f-039c0f41ee79.png)

4) Go to `B`
![selection_033](https://user-images.githubusercontent.com/18367406/42157064-04152d1e-7ded-11e8-9f24-098d91019809.png)

**A's filterDefaultValues were merged with B's**

* CodeSandbox 

Demo: https://w7r6ypn47k.codesandbox.io/#/posts
Code: https://codesandbox.io/s/w7r6ypn47k

(Click on `Comments`)
![selection_034](https://user-images.githubusercontent.com/18367406/42157358-c539149c-7ded-11e8-9d75-cc02e70de467.png)

**Environment**

* React-admin version: 2.0.4
* Browser: Chrome

**Comment by djhi** [comment]
@fzaninotto Maybe we should store list filters by resources ?

**Comment by fzaninotto** [comment]
Thanks for the codesandbox, it confirms the bug. Marking it as such.

@djhi you're right, obviously.

**Comment by djhi** [comment]
I thought we still had list params in a reducer shared among resources but no, list params are already by resource.

After some digging it seems the issue lies in the resource reducer initialization but I can't figure out why yet.

**Comment by AkselsLedins** [comment]
It would certainly be interesting to let users put filters values to `null` from a URL. 


instead of:
https://github.com/marmelab/react-admin/blob/a1b27c4c21d70e94954441b26ceeaf9e848034c6/packages/ra-core/src/controller/ListController.js#L145-L146

if we do it like this that would allow this behaviour

```
if (!query.filter.hasOwnProperty(name)) {
  query.filter[name] = filterDefaultValues[name];
} 
```

-----

Our use case: we have a `ReferenceManyField` with a `Preview` button which is a component redirecting the user to the `List` view with a certain filter (call it A). But the `filterDefaultValues` of the List are in conflict with filter A. The user needs to remove filter B to see the same results that in `ReferenceManyField`.

**Comment by djhi** [comment]
Fixed by #2019. Will be available in `2.1.3`
